### PR TITLE
Add support for Feature flags.

### PR DESCRIPTION
**Pull request by ildyria**
This pull request reuse a similar interface of [pennant](https://laravel.com/docs/10.x/pennant) but in a much lighter version.

:warning: For this specific one, we do not make use of Facade as this breaks the route creations during the initial loading and phpstan parsing.

Using Features will also ensure that flags are well defined, an exception will be thrown if a flag is not found in the configuration. As usual flags are usually mapped to an optional ENV variable. This will allow us to integrates different code path and functionalities without relying too heavily on databases queries.

### CODE REVIEW

File: app/Assets/Features.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+final class Features
+{
+	/**
+	 * Determine whether a feature is active.
+	 *
+	 * @param string $featureName to check
+	 *
+	 * @return bool is active
+	 */
+	public static function active(string $featureName): bool
```
Comment (hereldar) [inline app/Assets/Features.php:19]: Is common to add the `is` prefix the methods that return a boolean, but I don't know the code base of this project, so keep the cohesion with the rest of the code.

File: app/Assets/Features.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	 *
+	 * @template T
+	 *
+	 * @param string|array<int,string> $featureNames to check
+	 * @param T                        $valIfTrue    Value if we features are enabled
+	 * @param T                        $valIfFalse   Value if we features are disabled
+	 *
+	 * @return T
+	 */
+	public static function whenConst(string|array $featureNames, mixed $valIfTrue, mixed $valIfFalse): mixed
```
Comment (hereldar) [inline app/Assets/Features.php]: The `const` suffix confused me until I discovered the `when()` method.

You could merge them (detecting that the value is callable).

Another option might be to use a different word, such as "whether".

**Review by hereldar — COMMENTED** [review]
The changes look good for me. I have added a couple of comments, but they are just suggestions.

File: app/Assets/Features.php (hunk 1 above)
Comment (ildyria) [reply app/Assets/Features.php:19]: I do agree with you, but in this specific case, I was trying to keep the interface of penant. 
https://laravel.com/docs/10.x/pennant#checking-features
image

File: app/Assets/Features.php (hunk 2 above)
Comment (ildyria) [reply app/Assets/Features.php]: Good point.

Comment (ildyria) [reply app/Assets/Features.php]: Fixed: [faee879](https://github.com/LycheeOrg/Lychee/pull/2282/commits/faee87946b9a53277e45f18ed98da788bfdf3684)

**Review by hereldar — APPROVED** [review]

**Review by d7415 — APPROVED** [review]
Based mostly on @hereldar's review

**Pull request merged**